python/variable_creation.cpp

```diff
@@ -24,6 +24,18 @@ void init_creation(py::module &m) {
       },
       py::arg("dims"), py::arg("shape"), py::arg("unit") = units::one,
       py::arg("dtype") = py::none(), py::arg("with_variances") = std::nullopt);
+  m.def(
+      "zeros",
+      [](const std::vector<Dim> &dims, const std::vector<scipp::index> &shape,
+         const units::Unit &unit, const py::object &dtype,
+         const bool with_variances) {
+        const auto dtype_ = scipp_dtype(dtype);
+        py::gil_scoped_release release;
+        return variable::zeros(Dimensions(dims, shape), unit, dtype_,
+                               with_variances);
```

Comment:
Any reason for using this instead of the constructor? How can we create variables with empty strings or empty datasets now? What about zero-vectors, or zero-matrices? What about complex numbers once they are supported?

Reply:
These things would have to be handled by `variable::zeros`. But I am not sure that `zeros` is the right function for strings and datasets. `np.zeros` works for strings but should we support it the same way?

Reply:
You are right that it feels like odd naming, but is there any harm?

We don't necessarily need `variable::zeros`, could just use `makeVariable`?

Reply:
Removing `variable::zeros` because implementing it for Dataset would require going through the variable factory. But we might want to add it at some point if we want to create a value initialised variable based on a dtype.

Reply:
I implemented it using `makeVariable<T>(Dimensions{dims, shape}, unit, Values{})`. But for `dtype=sc.dtype.Variable`, this creates 'invalid variables'. And for `dtype=DataArray` it causes a segfault when you do anything with it except assign to it. In that sense, it is very similar to `sc.empty`. Is this a problem with `zeros` or with `DataArray`? I.e. should DataArray have a check for nullptr like Variable?

Reply:
Just don't support `dtype=Variable` and `dtype=DataArray` here?